R-package/R/utils.R

```diff
@@ -187,6 +187,18 @@ xgb.iter.eval <- function(booster_handle, watchlist, iter, feval = NULL) {
 # Helper functions for cross validation ---------------------------------------
 #
 
+# Possibly convert the labels into factors, depending on the objective.
+# The labels are converted into factors only when the given objective refers to the classification
+# or ranking tasks.
+convert.labels <- function(labels, objective_name) {
+  if (objective_name %in% c('binary:logistic', 'binary:logitraw', 'binary:hinge', 'multi:softmax',
+                            'multi:softprob', 'rank:pairwise', 'rank:ndcg', 'rank:map')) {
+    return(factor(labels))
```

Comment:
I think `as.factor()` would be better. It can be faster, especially if `labels` is already a factor when this is called.

From `?factor`

> as.factor coerces its argument to a factor. It is an abbreviated (sometimes faster) form of factor.

Reply:
Didn't know that! Thanks! I'll make the change.

Reply:
See my latest commit.